request-manager/request/resumer.go

```diff
@@ -0,0 +1,620 @@
+// Copyright 2018, Square, Inc.
+
+package request
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/rand"
+
+	log "github.com/Sirupsen/logrus"
+	myconn "github.com/go-mysql/conn"
+
+	jr "github.com/square/spincycle/job-runner"
+	"github.com/square/spincycle/proto"
+	"github.com/square/spincycle/request-manager/db"
+)
+
+type Resumer interface {
+	// Suspend marks a running request as suspended and saves the corresponding
+	// suspended job chain.
+	Suspend(requestId string, sjc proto.SuspendedJobChain) error
+
+	// ResumeAll tries to resume all the SJCs currently stored in the database.
+	ResumeAll()
+
+	// Resume tries to resume a single SJC given its id and a connection to the
+	// database. The SJC must be claimed (`rm_host` field for the SJC must be set
+	// to the hostname given when creating the Resumer) before calling Resume, or
+	// it will fail.
+	Resume(id string, conn *sql.Conn) error
```

Comment:
`conn *sql.Conn` is implementation detail, should be removed from interface. This would precluded a full mock (or, at least, the mock would simply ignore this arg).